src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -239,11 +239,12 @@ public RevisionsCollectionConfiguration GetRevisionsConfiguration(string collect
             if (Configuration == null)
                 return ConflictConfiguration.Default;
 
-            if (Configuration.Collections != null &&
-                Configuration.Collections.TryGetValue(collection, out RevisionsCollectionConfiguration configuration))
+            foreach (var col in Configuration.Collections.
+                Where(col => string.Equals(col.Key, collection, StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
This will have a severe performance impact. The 'Collections' dictionary needs to have a proper comparer. I think you need to use `IFillFromBlittableJson` and implement it in RevisionsConfiguration.